magento/magento2
# Update theme design after layout merge called get theme

### Description (*)

Plugin aroundGetDbUpdateString
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Widget/Model/ResourceModel/Layout/Plugin.php#L41

Widget update Layout called from plugin above
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Widget/Model/ResourceModel/Layout/Update.php#L63

plugin module widget `aroundGetDbUpdateString` now retrieve correct theme instance instead get theme instance from construction(when layout merge init theme) 
https://github.com/magento/magento2/blob/5c31d9aead63106ff72a22b50458ef2112ffdff6/lib/internal/Magento/Framework/View/Model/Layout/Merge.php#L235

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#30656

### Manual testing scenarios (*)

Pre-config
Setup theme config default for store => luma theme
Scenario1
Product named "Joust Duffle Bag"
1. Go to admin -> Change Schedule Design Update : New theme: Choose blank theme
2. Create widget show products list type for blank theme and display only in product detail for "Joust Duffle Bag"
2.1 Type: Catalog Products List
2.2 Design Package/Theme: Blank theme
2.3 Assign to Store Views: All Store Views
2.4 Layout Updates: All product types
Specific Products: choose "Joust Duffle Bag"
Contianer: Main Container Bottom
Widget options: Choose default to make sure all critieria meet and valid widget to show
3. Save widget
4. Clear cache
5. Go to homepage first (Verify theme blank applied) OK it's work fine
6. Go to frontend visit page product "Joust Duffle Bag". Theme blank applied ok and expect widget created previously should visible
7.After page loaded see results. Theme blank show but widget not visible

### Expected results
page details of `Joust Duffle Bag` should applied blank theme (theme blank) not luma. Set xdebug breakpoint may help for easier verify

Widget should show and
Theme blank applied (Correct display)

Widget Catalog Product List should show in area `Main Content Top` of page details `Joust Duffle Bag` only

### Actually results (before fixes)
Widget Catalog Product List not visible on details page of `Joust Duffle Bag` (luma theme instance still used)

Widget not show (because magento still get theme luma default from initialized, Actually in the update layout theme applied use should be blank)
Theme blank applied (Correct display)

### Questions or comments

CC: @ericmorand @ihor-sviziev @kandy @orlangur

Test
@engcom-Alfa @engcom-Charlie 
Need coverage with MFTF will be very good
Add test for plugin widget non-coverage by any tests
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Widget/Model/ResourceModel/Layout/Plugin.php

### Contribution checklist (*)
